test/test_model.py

```diff
@@ -270,6 +271,26 @@ def test_update_app_relation_data(
             rel.data[harness.model.app].update(*args, **kwargs)
             assert harness.get_relation_data(relation_id, harness.model.app) == {'foo': 'baz'}
 
+    def test_set_app_relation_data(self, harness: ops.testing.Harness[ops.CharmBase]):
+        relation_name = 'db1'
+        original_data = {'deleted': 'item', 'unchanged': 'item', 'changed': 'original-value'}
+        new_data = {'new': 'item', 'unchanged': 'item', 'changed': 'new-value'}
+        # TODO: break this down into individual test cases? parametrize test?
```

Comment:
I'd say individual cases probably, unless it would parametrize easily, but it doesn't seem so at a glance.